src/coreclr/src/vm/ceemain.cpp

```diff
@@ -961,9 +960,6 @@ void EEStartupHelper(COINITIEE fFlags)
         // Now we really have fully initialized the garbage collector
         SetGarbageCollectorFullyInitialized();
 
-        // This will allocate a handle, so do this after GC is initialized.
-        CastCache::Initialize();
```

Comment:
It seems too early to bind to managed types here, so I had to move initialization further to a point where we can load `CastCache` class but before the casting could be used.